modules/Bio/EnsEMBL/DBSQL/SliceAdaptor.pm

```diff
@@ -2728,4 +2728,115 @@ sub _build_circular_slice_cache {
 } ## end _build_circular_slice_cache
 
 
+
+=head2 _fetch_by_fuzzy_matching
+  Args       : $cs, $version, $seq_region_name
+  Example    : my $fuzzy_matched_name = $self->fetch_by_fuzzy_matching( $cs, $version, $seq_region_name );
+  Description: fetches all the fuzzy matches for a given seq_region_name when requested
+  Returntype : string
+  Exceptions : None
+  Caller     : general
+  Status     : (refactored from fetch_by_region)
+=cut
+
+sub _fetch_by_fuzzy_matching {
+
+  my ($self, $cs, $version, $seq_region_name, $sql, $constraint, $bind_params) = @_;
+
+  my $csa = $self->db->get_CoordSystemAdaptor();
+  # my @bind_params;
+  my $length;
+
+  # my $sql = sprintf( "SELECT sr.name, sr.seq_region_id, sr.length, sr.coord_system_id "
+  #                  . "FROM seq_region sr " );
+  # my $constraint = "AND sr.coord_system_id = ?";
+
+  print "Do fuzzy matching...\n";
+
+  # Do fuzzy matching, assuming that we are just missing a version
+  # on the end of the seq_region name.
+
+  my $sth =
+    $self->prepare( $sql . " WHERE sr.name LIKE ? " . $constraint );
+
+  $bind_params->[0] = [ sprintf( '%s.%%', $seq_region_name ), SQL_VARCHAR ];
+
+  my $pos = 0;
+  foreach my $param (@$bind_params) {
+    # print "++$pos, $param->[0], $param->[1]\n";
```

Comment:
do we need to keep the print statement for debugging?

Reply:
Removed.